interfaces/core.go

```diff
@@ -38,6 +38,22 @@ func (plug *Plug) Ref() PlugRef {
 	return PlugRef{Snap: plug.Snap.Name(), Name: plug.Name}
 }
 
+// Sanitize plug with a given snapd interface.
+func (plug *Plug) Sanitize(iface Interface) error {
+	if iface.Name() != plug.Interface {
+		return fmt.Errorf("cannot sanitize plug %q (interface %q) using interface %q",
+			plug.Ref(), plug.Interface, iface.Name())
+	}
+	type sanitizer interface {
+		SanitizePlug(plug *Plug) error
```

Comment:
I'd make these interfaces public, to better document the thing (much as for example `http.Flusher` is public to document that). It adds a path to discovery of this feature.